packages/api/src/external/commonwell/patient.ts

```diff
@@ -152,6 +152,77 @@ export async function create(
   }
 }
 
+export async function retryLinking(patient: Patient, facilityId: string): Promise<boolean> {
+  let commonWell: CommonWellAPI | undefined;
+  try {
+    const { debug } = Util.out(`CW retry linking - M patientId ${patient.id}`);
+
+    const { organization, facility } = await getPatientData(patient, facilityId);
+    const orgName = organization.data.name;
+    const orgOID = organization.oid;
+    const facilityNPI = facility.data["npi"] as string;
+
+    // Patients of cxs that not go through EC should have theis status undefined so they're not picked up later
+    // when we enable it
+    const cqLinkStatus = (await isEnhancedCoverageEnabledForCx(patient.cxId))
+      ? "unlinked"
+      : undefined;
+    const storeIds = getStoreIdsFn(patient.id, patient.cxId, cqLinkStatus);
+
+    commonWell = makeCommonWellAPI(orgName, oid(orgOID));
+
+    const commonwellData = getCWData(patient.data.externalData);
+
+    if (!commonwellData) {
+      const msg = `The patient does not have external data`;
+      debug(
+        `ERR - ${msg} - Patient not registered to commonwell - ` +
+          `respUpdate: ${JSON.stringify(commonwellData)}`
+      );
+      throw new Error(msg);
+    }
+
+    const commonwellPatientId = commonwellData?.patientId;
+    const queryMeta = organizationQueryMeta(orgName, { npi: facilityNPI });
+    const commonwellPatient = patientToCommonwell({ patient, orgName, orgOID });
+
+    const respGet = await commonWell.getPatient(queryMeta, commonwellPatientId);
+    debug(`resp updatePatient: `, JSON.stringify(respGet));
+    const patientRefLink = respGet._links?.self?.href;
+    if (!patientRefLink) {
+      const msg = `Could not determine the patient ref link`;
+      debug(
+        `ERR - ${msg} - Patient retry did not occur because failed to get refLink - ` +
+          `respUpdate: ${JSON.stringify(respGet)}`
+      );
+      await storeIds({ commonwellPatientId, status: "failed" });
+      throw new Error(msg);
+    }
+
+    await findOrCreatePersonAndLink({
```

Comment:
ditto: link in this context is not Metriport patient <> CW patient.

Reply:
I replaced with:
`const link = await commonWell.addPatientLink(queryMeta, personId, cwPatient._links.self.href);

    await setCommonwellId({
      patientId: patient.id,
      cxId: patient.cxId,
      commonwellPatientId: cwPatientId,
      commonwellPersonId: personId,
      commonwellStatus: "completed",
    });`